sky/skypilot_config.py

```diff
@@ -93,8 +93,8 @@ def get_nested(keys: Sequence[str], default_value: Any) -> Any:
     return curr
 
 
-def pop_nested(keys: Sequence[str]) -> Dict[str, Any]:
-    """Returns a deep-copied config with the nested key popped.
+def set_nested(keys: Sequence[str], value: Optional[Any]) -> Dict[str, Any]:
```

Comment:
Does Any include None?

Reply:
Done. Thanks!

Reply:
Ahh good catch, it should include None, removed `Optional`